code/datums/components/bloodysoles.dm

```diff
@@ -275,6 +277,14 @@
 	..()
 
 /datum/component/bloodysoles/feet/on_step_blood(datum/source, obj/effect/decal/cleanable/pool)
+	//SKYRAT EDIT ADDITION BEGIN - DIGI_BLOODSOLE
+	if(!bloody_feet)
+		var/mob/living/carbon/H = parent
+		if (DIGITIGRADE in H.dna.species.species_traits)
+			bloody_feet = mutable_appearance('modular_skyrat/modules/digi_bloodsole/icons/effects/blood.dmi', "shoeblood", SHOES_LAYER)
+		else
+			bloody_feet = mutable_appearance('icons/effects/blood.dmi', "shoeblood", SHOES_LAYER)
+	//SKYRAT EDIT ADDITION END
```

Comment:
Why was this moved from `Initialize()`?

Reply:
It gets called too soon in both initialize() and update_icon(), because it gets called before copy_to() where the digitigrade prefs are put onto the character, so putting it in on_step_blood() is the only option.